lib/Test/Stream/Formatter/TAP.pm

```diff
@@ -99,6 +116,193 @@ sub _autoflush {
     select $old_fh;
 }
 
+sub event_tap {
+    my $self = shift;
+    my ($e, $num) = @_;
+
+    my $converter = $CONVERTERS{ref($e)};
+
+    $num = undef if $self->{+NO_NUMBERS};
+
+    # Legacy Support for $e->to_tap
+    unless ($converter) {
```

Comment:
You reply indicates a misunderstanding. It makes no sense to write an event specific formatter. During any test run there should only ever be one formatter, and one instance of that formatter. All events go to that formatter and it is responsible for formatting them into output. By default a formatter should ignore events it does not understand. This is the design.